components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java

```diff
@@ -4749,14 +4749,26 @@ public static NativeObject jsFunction_getDocument(Context cx, Scriptable thisObj
         }
         String resource = (String) args[1];
         String tenantDomain = (String) args[0];
-        Map<String, Object> docResourceMap = APIUtil.getDocument(username, resource, tenantDomain);
+        boolean isTenantFlowStarted = false;
+        try {
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```

Comment:
Tenant flow start flow code was moved to APIUtil.getDocument() method. Same issue existed there and is fixed in #1494 (https://github.com/wso2/carbon-apimgt/commit/26d642b9148c5a2b553209df8060c4857400ee4f)